parserator_web/templates/parserator_web/index.html

```diff
@@ -32,14 +31,79 @@ <h4>Parsing results</h4>
           </tbody>
         </table>
       </div>
+      <div id="parse-error" class="alert alert-danger mt-2 text-center" style="display:none" role="alert">
+        Uh-oh -- The inputted address could not be parsed!
+      </div>
+      <div id="empty-error" class="alert alert-danger mt-2 text-center" style="display:none" role="alert">
+        Please provide an address to be parsed.
+      </div>
     </div>
   </div>
 </div>
 {% endblock %}
 
 {% block extra_js %}
   <script type="text/javascript">
-    /* TODO: Flesh this out to connect the form to the API and render results
-    in the #address-results div. */
+    // on user submit, send user input to backend for parsing
+    // on return, update webpage with parsed address 
+    $('form').submit(function(e){
+
+      // disable refresh on submit
+      e.preventDefault();
+
+      // clean up existing results from prior submits
+      // hide parse & empty string error in case it is 
+      //      being displayed from a prior failure
+      $('#parse-error').hide();
+      $('#empty-error').hide();
+      // remove table body and hide results
+      $('table tbody > tr').remove();
+      $('#address-results').hide();
+
+      // grab user input_string
+      var input_string = $('#address').val();
+
+      // only parse input if user inputted something
+      if (input_string != ""){
```

Comment:
If a user submits a string of spaces (eg, `'    '`), this is the result:

<img width="1140" alt="Screen Shot 2020-10-13 at 11 26 52 AM" src="https://user-images.githubusercontent.com/1094243/95888800-2b95aa80-0d47-11eb-9cd6-8b0ed2ff067f.png">

How can you handle that case so that this submit function treats it as an empty string?

Reply:
I would have never thought of that edge case, thanks! I added `.trim()` to when I capture the input. Commit is [here](https://github.com/noah-marcus/code-challenge/commit/f262a05fb22b33ec0e01f06ede8f2dd248cfa0ae).